llvm/lib/Transforms/Coroutines/CoroSplit.cpp

```diff
@@ -1575,6 +1650,36 @@ static bool hasCallsBetween(Instruction *Save, Instruction *ResumeOrDestroy) {
   return false;
 }
 
+// Check if await-suspend helper is "simple".
+// The conditions are:
+// 1. The return result is exactly coroutine frame parameter, passed to helper
+// 2. There are no calls between any of the returns and helper entry that could
+// resume or destroy it
+// FIXME: perform more sophisiticated analysis?
+static bool isSimpleHelper(CoroAwaitSuspendInst *AWS) {
```

Comment:
Why do we have to handle the case? I don't feel it is necessary. It looks like a minor optimization which don't cover a lot of case but require more work to do.

Reply:
This method was added so that previously available optimization is not lost after this change. If you really feel this is too minor, I will remove it.